spark/v3.1/spark/src/main/java/org/apache/iceberg/spark/source/SparkWrite.java

```diff
@@ -586,8 +607,8 @@ public DataWriter<InternalRow> createWriter(int partitionId, long taskId) {
     @Override
     public DataWriter<InternalRow> createWriter(int partitionId, long taskId, long epochId) {
       Table table = tableBroadcast.value();
-      PartitionSpec spec = table.spec();
       FileIO io = table.io();
+      PartitionSpec outputSpec = table.specs().get(outputSpecId);
```

Comment:
nit: Shall we still call it `spec` to reduce the amount of changes?

Reply:
Done.